File: 0141-linked-list-cycle/0141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {

        ListNode* slow = head;
        ListNode* fast = head;
        unordered_map<ListNode*, int> mp;

        while(fast != NULL){
            if (mp.find(fast->next) != mp.end()){
                return true;
            }
            mp[fast]++;
            fast = fast->next;
        }

        return false;

        
    }
};

File: 0141-linked-list-cycle/0141-linked-list-cycle.js
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {

    let slow = head
    let fast = head

   while(fast!= undefined && fast.next!=undefined)
    {
        slow = slow.next
        fast = fast.next.next
        if(slow==fast) return true
    }
    return false
};



